webapp/crashstats/libbugzilla.py

```diff
@@ -80,40 +106,55 @@ def crash_report_to_description(crash_report_url, processed_crash):
         lines.append("")
         lines.append(f"Reason: ```{processed_crash['reason']}```")
 
-    # If there's a java_stack_trace, add that; this is a big string blob, so we truncate
-    # it at 5,000 characters but don't otherwise format it
-    if processed_crash.get("java_stack_trace"):
```

Comment:
Previously, this would prefer `java_stack_trace` over `java_exception` or processed minidump values. We want it the other way around--processed minidump takes precedence, followed by `java_exception`, followed by `java_stack_trace` if the other two didn't produce anything.

We're phasing out `java_stack_trace`, so at some point, we can remove the `java_stack_trace` nonsense altogether.

Reply:
`java_stack_trace` is a big string blob that we can't do much with in Socorro. `java_exception` is a structure that's JSON-encoded in the crash annotations. Because it's structured, we don't have to deal with parsing it.

This is the tracker bug covering phasing out `JavaStackTrace` and its usage: https://bugzilla.mozilla.org/show_bug.cgi?id=1851903

This is the one for Fenix (I think): https://bugzilla.mozilla.org/show_bug.cgi?id=1792902

There's no decision brief. We've had `JavaStackTrace` for a long time and it has a lot of issues and prevents us from improving Fenix support in Socorro. Roger and I spec'd out `JavaException` which is based on the Sentry error event exception structure. He implemented that a while ago (I forget when it was) and the plan was to phase out `JavaStackTrace` in favor of `JavaException`.